test/Promise.test.js

```diff
@@ -211,6 +211,74 @@ describe ('Promise', function () {
     });
   });
 
+  describe('finally', function () {
+    it('should call finally when resolved', function(done) {
+      var isFullfilled = false;
+      var finallyRan = false;
+      var p = new Promise(function(resolve, reject) {
+        resolve(1); 
+      }).then(function(value) {
+        isFullfilled = true; 
+      }).finally(function(value) {
+        assert.ok(isFullfilled, "should call finally after resolve");
+        assert.equal(value, undefined);
+        finallyRan = true;
+      }).then(function() {
+        assert.equal(finallyRan, true);
+      });
+
+      assert.strictEqual(p.resolved, true);
+      assert.strictEqual(p.rejected, false);
+      assert.strictEqual(p.pending, false);
+
+      done();
+    });
+
+    it('should call finally when rejected', function(done) {
+      var isFullfilled = false;
+      var finallyRan = false;
+      var p = new Promise(function(_resolve, reject) {
+        reject(new Error('An error has occured')); 
+      }).catch(function(_err) {
+        isFullfilled = true;
+        return err;
+      }).finally(function(value) {
+        assert.ok(isFullfilled, "should call finally after reject");
+        assert.equal(value, undefined);
+        finallyRan = true;
+      });
+
+      assert.strictEqual(p.resolved, false);
+      assert.strictEqual(p.rejected, true);
+      assert.strictEqual(p.pending, false);
+
+      done();
+    });
+
+    it('should continue promsie chain from finally', function(done) {
+      var isFullfilled = false;
+      var finallyRan = false;
+      var p = new Promise(function(resolve, reject) {
+        resolve(); 
+      }).then(function () {
+        isFullfilled = true; 
+      }).finally(function(value) {
+        assert.ok(isFullfilled, "should call finally after resolve");
+        assert.equal(value, undefined);
+        finallyRan = true;
+      }).then(function() {
+        assert.equal(finallyRan, true);
```

Comment:
Maybe move this `assert.equal(finallyRan, true)` into the last `return p.then(...)` to have all these final tests in a single place?